### PR TITLE
Use `assert` statement instead of `self.assert*` methods

### PR DESCRIPTION
**Pull request by toslunar**

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py [hunk 1]
```diff
@@ -252,43 +252,43 @@ def test_random(self):
...
+        assert x.dtype == self.dtype
+        assert x.format == self.format
 
         y = getattr(sparse, self.random_method)(
             3, 4, density=0.1,
             format=self.format, dtype=self.dtype,
             random_state=1)
 
-        self.assertTrue((x.toarray() == y.toarray()).all())
+        assert (x.toarray() == y.toarray()).all()
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py:273]: How about using `testing.assert_array_equal`?
```suggestion
        testing.assert_array_equal(x.toarray(), y.toarray())
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py [hunk 2]
```diff
@@ -252,43 +252,43 @@ def test_random(self):
...
+        assert x.format == self.format
 
         state2 = cupy.random.RandomState(1)
         y = getattr(sparse, self.random_method)(
             3, 4, density=0.1,
             format=self.format, dtype=self.dtype,
             random_state=state2)
 
-        self.assertTrue((x.toarray() == y.toarray()).all())
+        assert (x.toarray() == y.toarray()).all()
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py:291]:
```suggestion
        testing.assert_array_equal(x.toarray(), y.toarray())
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_csr.py [hunk 3]
```diff
@@ -1506,15 +1506,15 @@ def test_csr(self):
              cupy.array([], 'i'),
              cupy.array([0], 'i')),
             shape=(0, 0), dtype='f')
-        self.assertTrue(sparse.isspmatrix_csr(x))
+        assert sparse.isspmatrix_csr(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_csr.py:1509]:
```suggestion
        assert sparse.isspmatrix_csr(x) is True
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_csr.py [hunk 4]
```diff
@@ -1506,15 +1506,15 @@ def test_csr(self):
...
 
     def test_csc(self):
         x = sparse.csr_matrix(
             (cupy.array([], 'f'),
              cupy.array([], 'i'),
              cupy.array([0], 'i')),
             shape=(0, 0), dtype='f')
-        self.assertFalse(sparse.isspmatrix_csc(x))
+        assert not sparse.isspmatrix_csc(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_csr.py:1517]:
```suggestion
        assert sparse.isspmatrix_csc(x) is False
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_dia.py [hunk 5]
```diff
@@ -339,12 +339,12 @@ def test_dia(self):
             (cupy.array([], 'f'),
              cupy.array([0], 'i')),
             shape=(0, 0), dtype='f')
-        self.assertTrue(sparse.isspmatrix_dia(x))
+        assert sparse.isspmatrix_dia(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_dia.py:342]:
```suggestion
        assert sparse.isspmatrix_dia(x) is True
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_dia.py [hunk 6]
```diff
@@ -339,12 +339,12 @@ def test_dia(self):
...
 
     def test_csr(self):
         x = sparse.csr_matrix(
             (cupy.array([], 'f'),
              cupy.array([], 'i'),
              cupy.array([0], 'i')),
             shape=(0, 0), dtype='f')
-        self.assertFalse(sparse.isspmatrix_dia(x))
+        assert not sparse.isspmatrix_dia(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_dia.py:350]:
```suggestion
        assert sparse.isspmatrix_dia(x) is False
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py [hunk 7]
```diff
@@ -1023,12 +1023,12 @@ def test_coo(self):
             (cupy.array([0], 'f'),
              (cupy.array([0], 'i'), cupy.array([0], 'i'))),
             shape=(1, 1), dtype='f')
-        self.assertTrue(sparse.isspmatrix_coo(x))
+        assert sparse.isspmatrix_coo(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py:1026]:
```suggestion
        assert sparse.isspmatrix_coo(x) is True
```

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py [hunk 8]
```diff
@@ -1023,12 +1023,12 @@ def test_coo(self):
...
 
     def test_csr(self):
         x = sparse.csr_matrix(
             (cupy.array([], 'f'),
              cupy.array([], 'i'),
              cupy.array([0], 'i')),
             shape=(0, 0), dtype='f')
-        self.assertFalse(sparse.isspmatrix_coo(x))
+        assert not sparse.isspmatrix_coo(x)
```
Comment (asi1024) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_coo.py:1034]:
```suggestion
        assert sparse.isspmatrix_coo(x) is False
```

File: tests/cupy_tests/core_tests/test_internal.py [hunk 9]
```diff
@@ -55,64 +55,62 @@ def test_invalid(self):
 class TestVectorEqual(unittest.TestCase):
 
     def test_empty(self):
-        self.assertEqual(internal.vector_equal([], []), True)
+        assert internal.vector_equal([], []) is True
```
Comment (emcastillo) [inline tests/cupy_tests/core_tests/test_internal.py:58]:
```suggestion
        assert internal.vector_equal([], [])
```

File: tests/cupy_tests/core_tests/test_internal.py [hunk 10]
```diff
@@ -55,64 +55,62 @@ def test_invalid(self):
...
 
     def test_not_equal(self):
-        self.assertEqual(internal.vector_equal([1, 2, 3], [1, 2, 0]), False)
+        assert internal.vector_equal([1, 2, 3], [1, 2, 0]) is False
```
Comment (emcastillo) [inline tests/cupy_tests/core_tests/test_internal.py:61]:
```suggestion
        assert not internal.vector_equal([1, 2, 3], [1, 2, 0])
```

File: tests/cupy_tests/core_tests/test_internal.py [hunk 11]
```diff
@@ -55,64 +55,62 @@ def test_invalid(self):
...
 
     def test_equal(self):
-        self.assertEqual(internal.vector_equal([-1, 0, 1], [-1, 0, 1]), True)
+        assert internal.vector_equal([-1, 0, 1], [-1, 0, 1]) is True
```
Comment (emcastillo) [inline tests/cupy_tests/core_tests/test_internal.py:64]:
```suggestion
        assert internal.vector_equal([-1, 0, 1], [-1, 0, 1])
```

File: tests/cupy_tests/core_tests/test_internal.py [hunk 12]
```diff
@@ -55,64 +55,62 @@ def test_invalid(self):
...
 
     def test_different_size(self):
-        self.assertEqual(internal.vector_equal([1, 2, 3], [1, 2]), False)
+        assert internal.vector_equal([1, 2, 3], [1, 2]) is False
```
Comment (emcastillo) [inline tests/cupy_tests/core_tests/test_internal.py:67]:
```suggestion
        assert not internal.vector_equal([1, 2, 3], [1, 2])
```

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by toslunar** [comment]
pfnCI, test this please.

**Pull request merged**